### DISCUSSION TITLE
weird Losses and accuracies

### DISCUSSION THREAD

**Discussion by supersyz** [fvmassoli/cross-resolution-face-recognition#8] [General]
Hi，thx for your previous help ,
I tried SGD again,the accuracy seems to stuck at 20%, And All losses begin to grow with accuracy decreasing.
I don't know if it's because lack of training or something wrong. Could you tell me how many epoch did you train the model well?

And by the way,did you succeeded by using default hyper parameters? If so,maybe I made the data set wrong,because I can not.
But it seems to be much better when I adjust learning rate to 0.0001.

Specifically,
1.I download VGGFace2 data set and upzid.
2.Then I write a program to make validation set（first make directory 'validation' ） :
`# -*- coding: utf-8 -*-

import os,shutil,random

file = r"./train"
files = os.listdir(file)

os.chdir(file)

print(os.getcwd())
# print(files[0])
for i in files:

    subfile = os.path.join(os.getcwd() + "/" , i)

    # print("subfile:",subfile)
    subfiles = os.listdir(subfile)
    a = random.sample(subfiles,10)
    if not os.path.exists(os.getcwd()[:-5] + '' + 'validation' + i):
        os.mkdir(os.getcwd()[:-5] + '' + 'validation' + '/' + i)
        # print(a)
    for k in a:

        finalfile = os.path.join(subfile + "/", k)
        # print(finalfile)
        shutil.move(finalfile, os.getcwd()[:-5]+''+'validation'+'/'+i)
        # finalfile = os.path.join(os.getcwd() + "\\" , i)

`
Then the validation set has the same number of directories as train set with the same name.the difference is the validation image in each ID is 10 images from train set ID

3.I have 2 GPUs so in class Trainer, I set  
`        self._student = torch.nn.DataParallel(student,device_ids=[2, 3])
        self._teacher = torch.nn.DataParallel(teacher,device_ids=[2 3])
`
4.Then I begin to train with defalut setting  but the losses expand.
Then I set batchsize=64 learning-rate=0.0001  currirulum learning  nesterov   ,losses seem to converge.
but the valid acc LR seems to stuck at around 20%, And All losses then began to grow with accuracy decreasing.
I don't know if it's because lack of training or something wrong. Could you tell me how many epoch did you train the model well?

The train loss is so strange that it decrease at first and then increase till almost same as first train.when the next epoch comes,it suddenly becomes very low and then slightly goes up again. 

And by the way,did you succeeded by using default hyper parameters? If so,maybe I made the data set wrong,because I can not.
But it seems to be much better when I adjust learning rate to 0.0001.

I think maybe it's because my data set was made wrong,but I can't figure it out.

**Comment by fvmassoli** [comment]
The training details are reported in the published version of the paper so you can look at it for the details.
If our hyperparameters do not work in your case might be due to several reasons, different hardware, python/pytorch version, different seed initialization, etc...
You should look more carefully at your training while training, for example you monitor its behaviour more frequently and try to spot overfitting problems. Or perhaps if it worked better with lower learning rate then you it.

**Reply by supersyz** [reply]
Thx a lot,I'm going to do more experiments and find the hyper parameters work for me

**Reply by supersyz** [reply]
Hi, After doing some adjustment of hyper parameters,I got some better but still not satisfied models,could you pull the final model you trained well ? I'd like to some some further research on that   :)  thx a lot

**Reply by fvmassoli** [reply]
Can you tell me what kind of experiments did you do? With which hyperparameters? What were the performance of your mdoels?

I cannot send to you the model checkpoint since the project is partially funded by industries.

**Reply by supersyz** [reply]
YES.absolutely.
I tried the paper hyper parameters,but maybe due to the fact that my device cannot afford the batch in the paper（256），It didn't work.
My adjustment is as follows:
1.
`        teacher_features, teacher_logits = self._teacher(batch_original.to(self._device))
        student_features, student_logits = self._student(batch.to(self._device))
`
to use two GPUS to run the model
`optimizer = Adam(params=sm.parameters(),
lr=args.learning_rate,weight_decay=3e-04)`

`
scheduler = ReduceLROnPlateau( 
                        optimizer=optimizer, 
                        mode='min', 
                        factor=0.95, 
                        patience=args.scheduler_patience, 
                        verbose=True, 
                        min_lr=0, 
                        threshold=0.1 
                    ) 

`


to change the optimizer and scheduler.

3.  try batchsize=64 and initial learning-rate  0.0006.  The performances of Training Stage and Validation HR are quite good around 90%（about 94% at the peak） but the validation LR only reached 55% at the peak. I think maybe finetuning more will help.

I'm not in a certain company and need to use it for some commercial usage,but I'm a master student in University of Chinese Academy of Sciences.As your paper says,you focus more on cross resolution face images,rather than only low resolution face images.So I was just curious about its application for only LR images and of course, it would be great if I can make some progress and publish a further paper.  :)  

if you need any official email of my institution, plz let me know anything you nedd    :)

**Reply by supersyz** [reply]
（I'm not in a certain company and won't use it for some commercial usage，just personal study）
